### PR TITLE
update our product build matrix

### PR DESCRIPTION
**Pull request by devoncarew**
- update our product build matrix
- support 2017.1
- initial support for separate sdks per project

fix https://github.com/flutter/flutter-intellij/issues/823

### CODE REVIEW

**Comment by devoncarew** [comment]
@pq @skybrian

**Comment by pq** [comment]
So awesome.  LGTM! 🤘

File: src/io/flutter/dart/DartSdkLibUtilDelegate.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public void ensureDartSdkConfigured(@Nullable Project project, String sdkHomePat
       }
       else {
         //noinspection unchecked
-        final Method method = delegate.getMethod("ensureDartSdkConfigured", Project.class, Collection.class);
+        final Method method = delegate.getMethod("ensureDartSdkConfigured", Project.class, String.class);
```
Comment (pq) [inline src/io/flutter/dart/DartSdkLibUtilDelegate.java]: Do we not need to support both APIs?

Comment (devoncarew) [inline src/io/flutter/dart/DartSdkLibUtilDelegate.java]: This was an error on my part in the method params. Pre 2017.1, this method in the dart plugin took one param; in 2017.1+ it takes two.

**Pull request merged**